builtin/credential/approle/path_role.go

```diff
@@ -550,22 +550,27 @@ func (b *backend) pathRoleSecretIDList(req *logical.Request, data *framework.Fie
 		// corresponding lock many times using SecretIDs is not
 		// possible. Also, indexing it everywhere using secretIDHMACs
 		// makes listing operation easier.
-		lock := b.secretIDLock(secretIDHMAC)
-		lock.RLock()
+		secretIDLock := b.secretIDLock(secretIDHMAC)
```

Comment:
Yeah. Essentially it is the case where secretIDHMAC == "". But secretIDHMAC is always a valid hex encoded string. So hitting this case is also unlikely. I wasn't able to reason when and how the locks will be same. Also, I can't see a case where this ends up with a deadlock when locks are not the same.